# Mathematical Functions (math Module)
- A module is collection of functions,variables and classes etc.
- library is ntg but grp of modules(library is high level component).
- math is a module that contains several functions to perform mathematical operations.

- if we want to use any module in python,by default it is available to python but not for our program..compulsory we have to import that module.

In [2]:
print(math.sqrt(16))

NameError: name 'math' is not defined

In [3]:
import math 
print(math.sqrt(16))#sqrt is function
print(math.pi)
print(math.e)

4.0
3.141592653589793
2.718281828459045


- We can create alias name by using as keyword.
- once we create alias name,by using that we can access functions and variables of that module.

In [1]:
import math as m
print(m.sqrt(36))
print(math.sqrt(25))

6.0


NameError: name 'math' is not defined

- we can import a particular member of a module explicitly as follows
  - from math import sqrt
  - from math import sqrt,pi
- if we import a member explicity then it is not required to use module name while accessing.

In [3]:
from math import sqrt,pi
print(sqrt(16))
print(pi)

4.0
3.141592653589793


In [5]:
#This type is not recommended..redability is down
 
from math import * #implicit import
print(sqrt(25))
print(pi)

5.0
3.141592653589793


## important functions of math module
- sqrt()
- ceil()
- floor()
- pow(x,y)
- factorial()
- trunc(x)
- gcd(x)
- sin(x
- cos(x)
- tan(x)

### Important variables of math module or These are various constants  present in math module
- pi=3.14
- e=2.71
- inf=infinity
- nan=not a number

In [9]:
#Write a python program to find area of circle
from math import pi
r=int(input("Enter a radius value:"))
print("Area of circle:",pi*r**2)

Enter a radius value:5
Area of circle: 78.53981633974483


### Difference b/w  import math and from math import *

    import math
    
- while accessing the members of module we can use 
- print(math.sqrt(4))

 from math import *
 
- while accesing the members of module we can use 
    - print(sqrt(4))